src/canopy_cli/data_loader/data_loader.py

```diff
@@ -61,10 +57,20 @@ def _df_to_documents(df: pd.DataFrame) -> List[Document]:
     try:
         if "metadata" in df.columns:
             df.loc[:, "metadata"] = df["metadata"].apply(_process_metadata)
-        documents = [
-            Document(**{k: v for k, v in row._asdict().items() if not pd.isna(v)})
-            for row in df.itertuples(index=False)
-        ]
+        documents = []
+        for row in df.itertuples(index=False):
+            try:
+                documents.append(
+                    Document(
+                        **{k: v for k, v in row._asdict().items() if not pd.isna(v)}
+                    )
+                )
+            except ValidationError as e:
```

Comment:
If you don't have a dedicated handling of `ValidationError` which is different than how you'd handle other error types - just stick with `except Exception as e`.  
Otherwise we might miss certain Exceptions, and they will halt execution ungracefully.

Reply:
I intended other errors that are not validation to be raised as is (and with time handle them) note that the only think we are trying is to construct a document. does it make sense?